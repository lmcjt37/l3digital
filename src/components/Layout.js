import React from 'react'
import PropTypes from 'prop-types'

import Header from './Header'
import Menu from './Menu'
import Contact from './Contact'
import Footer from './Footer'
import Transitions from './Transitions'

import 'assets/scss/main.scss'

class Layout extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            isMenuVisible: false,
            loading: 'is-loading',
        }
        this.handleToggleMenu = this.handleToggleMenu.bind(this)
    }

    componentDidMount() {
        this.timeoutId = setTimeout(() => {
            this.setState({ loading: '' })
        }, 100)
    }

    componentWillUnmount() {
        if (this.timeoutId) {
            clearTimeout(this.timeoutId)
        }
    }

    handleToggleMenu() {
        this.setState({
            isMenuVisible: !this.state.isMenuVisible,
        })
    }

    render() {
        const { children, location } = this.props

        return (
            <div
                className={`body ${this.state.loading} ${
                    this.state.isMenuVisible ? 'is-menu-visible' : ''
                }`}
            >
                <div id="wrapper">
                    <Header onToggleMenu={this.handleToggleMenu} />

                    <Transitions location={location}>{children}</Transitions>

                    <Contact />
                    <Footer />
                </div>
                <Menu onToggleMenu={this.handleToggleMenu} />
            </div>
        )
    }
}

Layout.propTypes = {
    children: PropTypes.object,
    location: PropTypes.object,
}

export default Layout
