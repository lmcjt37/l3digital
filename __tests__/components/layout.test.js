import React from 'react'
import renderer from 'react-test-renderer'
import { StaticQuery } from 'gatsby'

import Layout from '../../src/components/Layout'
import contactMock from '../__mocks__/contact.mock'

beforeEach(() => {
    StaticQuery.mockImplementationOnce(({ render }) => render(contactMock))
})

describe('Layout tests', () => {
    it('check it renders correctly with snapshot', () => {
        const tree = renderer.create(<Layout />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
