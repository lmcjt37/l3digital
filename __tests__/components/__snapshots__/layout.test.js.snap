// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout tests check it renders correctly with snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="body is-loading "
      >
        <div
          id="wrapper"
        >
          <header
            class="alt"
            id="header"
          >
            <a
              class="logo"
              href="/"
              partiallyactive="false"
              state="[object Object]"
            >
              <svg
                version="1.1"
                viewBox="15 5 70 90"
              >
                <path
                  d="M42.487,89.021c0.259,0,0.519-0.067,0.751-0.201l16.223-9.376l16.238-9.365c0.464-0.269,0.75-0.764,0.75-1.3 V50.043c0-0.536-0.286-1.031-0.75-1.299l-15.489-8.94V21.94c0-0.536-0.286-1.031-0.749-1.299l-16.223-9.376 c-0.232-0.134-0.492-0.201-0.751-0.201s-0.519,0.067-0.751,0.201l-16.222,9.376c-0.464,0.268-0.749,0.763-0.749,1.299v18.73v18.739 v18.736c0,0.535,0.285,1.03,0.749,1.299l16.222,9.376C41.969,88.954,42.229,89.021,42.487,89.021z M74.449,68.49L59.21,77.279 V59.698l15.239-8.789V68.49z M58.21,59.121l-15.223,8.792V50.331l15.223-8.795V59.121z M42.987,69.067l15.223-8.792v17.582 l-15.223,8.798V69.067z M58.21,40.382l-15.223,8.795V31.595l15.223-8.789V40.382z M41.987,31.595v17.582l-15.222-8.795V22.806 L41.987,31.595z M41.987,50.331v17.582l-15.222-8.792V41.536L41.987,50.331z M26.766,60.275l15.222,8.792v17.588l-15.222-8.798 V60.275z M 86.75, 73.5 m -40, 0 a 2,4 20 0,0 8,0 a 2,4 20 0,0 -8,0 M 86.75, 54.5 m -40, 0 a 2,4 20 0,0 8,0 a 2,4 20 0,0 -8,0 M 86.75, 35.5 m -40, 0 a 2,4 20 0,0 8,0 a 2,4 20 0,0 -8,0 "
                />
              </svg>
               
              <span>
                L3 Digital
              </span>
            </a>
            <nav>
              <a
                class="menu-link"
                href="javascript:;"
              >
                Menu
              </a>
            </nav>
          </header>
          <div>
            <div
              style="transition: transform 1125ms ease, opacity 750ms ease; transform: scale(1) translateZ(0); opacity: 1;"
            />
          </div>
          <section
            id="contact"
          >
            <div
              class="inner"
            >
              <section>
                <form
                  action="#"
                  method="post"
                >
                  <div
                    class="field half first"
                  >
                    <label
                      for="name"
                    >
                      Name
                    </label>
                    <input
                      id="name"
                      name="name"
                      type="text"
                    />
                  </div>
                  <div
                    class="field half"
                  >
                    <label
                      for="email"
                    >
                      Email
                    </label>
                    <input
                      id="email"
                      name="email"
                      type="text"
                    />
                  </div>
                  <div
                    class="field"
                  >
                    <label
                      for="message"
                    >
                      Message
                    </label>
                    <textarea
                      id="message"
                      name="message"
                      rows="6"
                    />
                  </div>
                  <ul
                    class="actions"
                  >
                    <li>
                      <input
                        class="special"
                        type="submit"
                        value="Send Message"
                      />
                    </li>
                    <li>
                      <input
                        type="reset"
                        value="Clear"
                      />
                    </li>
                  </ul>
                </form>
              </section>
              <section
                class="split"
              >
                <section>
                  <div
                    class="contact-method"
                  >
                    <span
                      class="icon alt fa-envelope"
                    />
                    <h3>
                      Email
                    </h3>
                    <a
                      href="mailto:hello@l3digital.co.uk"
                    >
                      hello@l3digital.co.uk
                    </a>
                  </div>
                </section>
                <section>
                  <div
                    class="contact-method"
                  >
                    <span
                      class="icon alt fa-phone"
                    />
                    <h3>
                      Phone
                    </h3>
                    <a
                      href="tel:0845 111 3333"
                    >
                      0845 111 3333
                    </a>
                  </div>
                </section>
                <section>
                  <div
                    class="contact-method"
                  >
                    <span
                      class="icon alt fa-home"
                    />
                    <h3>
                      Address
                    </h3>
                    <span>
                      L3 Digital Limited
                      <br />
                      Amazing House
                      <br />
                      Leicestershire
                      <br />
                      United Kingdom
                      <br />
                      LE3 T03
                    </span>
                  </div>
                </section>
              </section>
            </div>
          </section>
          <footer
            id="footer"
          >
            <div
              class="inner"
            >
              <ul
                class="icons"
              >
                <li>
                  <a
                    aria-label="Twitter"
                    class="icon alt fa-twitter"
                    href="https://twitter.com/l3digital_uk"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <span
                      class="label"
                    >
                      Twitter
                    </span>
                  </a>
                </li>
                <li>
                  <a
                    aria-label="Github"
                    class="icon alt fa-github"
                    href="https://github.com/l3digitaluk"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <span
                      class="label"
                    >
                      GitHub
                    </span>
                  </a>
                </li>
                <li>
                  <a
                    aria-label="LinkedIn"
                    class="icon alt fa-linkedin"
                    href="https://www.linkedin.com/in/l3-digital-uk/"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <span
                      class="label"
                    >
                      LinkedIn
                    </span>
                  </a>
                </li>
              </ul>
              <ul
                class="copyright"
              >
                <li>
                  Â© 
                  2019
                   L3 Digital
                </li>
                <li>
                  All Rights Reserved
                </li>
              </ul>
            </div>
          </footer>
        </div>
        <nav
          id="menu"
        >
          <div
            class="inner"
          >
            <ul
              class="links"
            >
              <li>
                <a
                  activeclassname="active-effect"
                  class="link-effect"
                  href="/"
                  partiallyactive="false"
                  state="[object Object]"
                >
                  Home
                </a>
              </li>
              <li>
                <a
                  activeclassname="active-effect"
                  class="link-effect"
                  href="/projects"
                  partiallyactive="true"
                  state="[object Object]"
                >
                  Projects
                </a>
              </li>
              <li>
                <a
                  activeclassname="active-effect"
                  class="link-effect"
                  href="/about"
                  partiallyactive="false"
                  state="[object Object]"
                >
                  About Us
                </a>
              </li>
            </ul>
            <ul
              class="icons"
            >
              <li>
                <a
                  aria-label="Twitter"
                  class="icon alt fa-twitter"
                  href="https://twitter.com/l3digital_uk"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <span
                    class="label"
                  >
                    Twitter
                  </span>
                </a>
              </li>
              <li>
                <a
                  aria-label="Github"
                  class="icon alt fa-github"
                  href="https://github.com/l3digitaluk"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <span
                    class="label"
                  >
                    GitHub
                  </span>
                </a>
              </li>
              <li>
                <a
                  aria-label="LinkedIn"
                  class="icon alt fa-linkedin"
                  href="https://www.linkedin.com/in/l3-digital-uk/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <span
                    class="label"
                  >
                    LinkedIn
                  </span>
                </a>
              </li>
            </ul>
          </div>
          <a
            class="close"
            href="javascript:;"
          >
            Close
          </a>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="body is-loading "
    >
      <div
        id="wrapper"
      >
        <header
          class="alt"
          id="header"
        >
          <a
            class="logo"
            href="/"
            partiallyactive="false"
            state="[object Object]"
          >
            <svg
              version="1.1"
              viewBox="15 5 70 90"
            >
              <path
                d="M42.487,89.021c0.259,0,0.519-0.067,0.751-0.201l16.223-9.376l16.238-9.365c0.464-0.269,0.75-0.764,0.75-1.3 V50.043c0-0.536-0.286-1.031-0.75-1.299l-15.489-8.94V21.94c0-0.536-0.286-1.031-0.749-1.299l-16.223-9.376 c-0.232-0.134-0.492-0.201-0.751-0.201s-0.519,0.067-0.751,0.201l-16.222,9.376c-0.464,0.268-0.749,0.763-0.749,1.299v18.73v18.739 v18.736c0,0.535,0.285,1.03,0.749,1.299l16.222,9.376C41.969,88.954,42.229,89.021,42.487,89.021z M74.449,68.49L59.21,77.279 V59.698l15.239-8.789V68.49z M58.21,59.121l-15.223,8.792V50.331l15.223-8.795V59.121z M42.987,69.067l15.223-8.792v17.582 l-15.223,8.798V69.067z M58.21,40.382l-15.223,8.795V31.595l15.223-8.789V40.382z M41.987,31.595v17.582l-15.222-8.795V22.806 L41.987,31.595z M41.987,50.331v17.582l-15.222-8.792V41.536L41.987,50.331z M26.766,60.275l15.222,8.792v17.588l-15.222-8.798 V60.275z M 86.75, 73.5 m -40, 0 a 2,4 20 0,0 8,0 a 2,4 20 0,0 -8,0 M 86.75, 54.5 m -40, 0 a 2,4 20 0,0 8,0 a 2,4 20 0,0 -8,0 M 86.75, 35.5 m -40, 0 a 2,4 20 0,0 8,0 a 2,4 20 0,0 -8,0 "
              />
            </svg>
             
            <span>
              L3 Digital
            </span>
          </a>
          <nav>
            <a
              class="menu-link"
              href="javascript:;"
            >
              Menu
            </a>
          </nav>
        </header>
        <div>
          <div
            style="transition: transform 1125ms ease, opacity 750ms ease; transform: scale(1) translateZ(0); opacity: 1;"
          />
        </div>
        <section
          id="contact"
        >
          <div
            class="inner"
          >
            <section>
              <form
                action="#"
                method="post"
              >
                <div
                  class="field half first"
                >
                  <label
                    for="name"
                  >
                    Name
                  </label>
                  <input
                    id="name"
                    name="name"
                    type="text"
                  />
                </div>
                <div
                  class="field half"
                >
                  <label
                    for="email"
                  >
                    Email
                  </label>
                  <input
                    id="email"
                    name="email"
                    type="text"
                  />
                </div>
                <div
                  class="field"
                >
                  <label
                    for="message"
                  >
                    Message
                  </label>
                  <textarea
                    id="message"
                    name="message"
                    rows="6"
                  />
                </div>
                <ul
                  class="actions"
                >
                  <li>
                    <input
                      class="special"
                      type="submit"
                      value="Send Message"
                    />
                  </li>
                  <li>
                    <input
                      type="reset"
                      value="Clear"
                    />
                  </li>
                </ul>
              </form>
            </section>
            <section
              class="split"
            >
              <section>
                <div
                  class="contact-method"
                >
                  <span
                    class="icon alt fa-envelope"
                  />
                  <h3>
                    Email
                  </h3>
                  <a
                    href="mailto:hello@l3digital.co.uk"
                  >
                    hello@l3digital.co.uk
                  </a>
                </div>
              </section>
              <section>
                <div
                  class="contact-method"
                >
                  <span
                    class="icon alt fa-phone"
                  />
                  <h3>
                    Phone
                  </h3>
                  <a
                    href="tel:0845 111 3333"
                  >
                    0845 111 3333
                  </a>
                </div>
              </section>
              <section>
                <div
                  class="contact-method"
                >
                  <span
                    class="icon alt fa-home"
                  />
                  <h3>
                    Address
                  </h3>
                  <span>
                    L3 Digital Limited
                    <br />
                    Amazing House
                    <br />
                    Leicestershire
                    <br />
                    United Kingdom
                    <br />
                    LE3 T03
                  </span>
                </div>
              </section>
            </section>
          </div>
        </section>
        <footer
          id="footer"
        >
          <div
            class="inner"
          >
            <ul
              class="icons"
            >
              <li>
                <a
                  aria-label="Twitter"
                  class="icon alt fa-twitter"
                  href="https://twitter.com/l3digital_uk"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <span
                    class="label"
                  >
                    Twitter
                  </span>
                </a>
              </li>
              <li>
                <a
                  aria-label="Github"
                  class="icon alt fa-github"
                  href="https://github.com/l3digitaluk"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <span
                    class="label"
                  >
                    GitHub
                  </span>
                </a>
              </li>
              <li>
                <a
                  aria-label="LinkedIn"
                  class="icon alt fa-linkedin"
                  href="https://www.linkedin.com/in/l3-digital-uk/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <span
                    class="label"
                  >
                    LinkedIn
                  </span>
                </a>
              </li>
            </ul>
            <ul
              class="copyright"
            >
              <li>
                Â© 
                2019
                 L3 Digital
              </li>
              <li>
                All Rights Reserved
              </li>
            </ul>
          </div>
        </footer>
      </div>
      <nav
        id="menu"
      >
        <div
          class="inner"
        >
          <ul
            class="links"
          >
            <li>
              <a
                activeclassname="active-effect"
                class="link-effect"
                href="/"
                partiallyactive="false"
                state="[object Object]"
              >
                Home
              </a>
            </li>
            <li>
              <a
                activeclassname="active-effect"
                class="link-effect"
                href="/projects"
                partiallyactive="true"
                state="[object Object]"
              >
                Projects
              </a>
            </li>
            <li>
              <a
                activeclassname="active-effect"
                class="link-effect"
                href="/about"
                partiallyactive="false"
                state="[object Object]"
              >
                About Us
              </a>
            </li>
          </ul>
          <ul
            class="icons"
          >
            <li>
              <a
                aria-label="Twitter"
                class="icon alt fa-twitter"
                href="https://twitter.com/l3digital_uk"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  class="label"
                >
                  Twitter
                </span>
              </a>
            </li>
            <li>
              <a
                aria-label="Github"
                class="icon alt fa-github"
                href="https://github.com/l3digitaluk"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  class="label"
                >
                  GitHub
                </span>
              </a>
            </li>
            <li>
              <a
                aria-label="LinkedIn"
                class="icon alt fa-linkedin"
                href="https://www.linkedin.com/in/l3-digital-uk/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  class="label"
                >
                  LinkedIn
                </span>
              </a>
            </li>
          </ul>
        </div>
        <a
          class="close"
          href="javascript:;"
        >
          Close
        </a>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
